docs/DeveloperGuide.md

```diff
@@ -330,30 +330,318 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+**Use case: UC06 Mark frequently used contacts**
+
+**MSS**
+
+1.  User requests to list the contacts or [search (UC02)](#uc02) for a contact
+2.  ContactCS shows a list of contacts
+3.  User requests to mark certain contacts as frequently accessed
+4.  ContactCS removes the marking from these contacts for quick access
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a. The list is empty.
+    * 1a1. ContactCS displays a message to the user saying that the list is empty
 
-  Use case ends.
+      Use case ends.
 
 * 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. ContactCS requests the user to provide the correct index
+    * 3a2. User enters new index
+    * Steps 3a1 - 3a2 are repeated until the index entered are correct
+
+      Use case resumes from step 4.
+
+* 3b. The given format is invalid.
+
+    * 3b1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 3b2. User enters new input
+    * Steps 3b1 - 3b2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 4.
+
+**Use case: UC07 Access frequently used contacts**
+
+**MSS**
+
+1.  User requests to access frequently used contacts
+2.  ContactCS shows a list of matching contacts
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The list is empty.
+  * 1a1. ContactCS displays a message to the user saying that the list is empty
+
+    Use case ends.
+
+* 1b. The given format is invalid.
+
+    * 1b1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 1b2. User enters new input
+    * Steps 1b1 - 1b2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 2.
+
+**Use case: UC08 View usage instructions**
+
+**MSS**
+
+1.  User requests to view usage instructions for the app
+2.  ContactCS displays the usage instructions and commands available for the user
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The given format is invalid.
+
+    * 1a1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 1a2. User enters new input
+    * Steps 1a1 - 1a2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 2.
+
+**Use case: UC09 View app with sample data**
+
+**MSS**
+
+1.  User requests to view app with sample data
+2.  ContactCS populates the interface with sample data for demonstration purposes
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The given format is invalid.
+
+    * 1a1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 1a2. User enters new input
+    * Steps 1a1 - 1a2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 2.
+
+<a name="uc10"></a>
+
+**Use case: UC10 Undo actions**
+
+**MSS**
+
+1.  User performs an action in ContactCS
```

Comment:
I am not sure if step 1 is supposed to be here because technically speaking, it is not part of this use case; also, you have taken care of the situation where no operation has been performed yet so there's no need to make sure there is one.

Same applies to line 625.

Reply:
Will remove it in the next commit.